code/clue.js

```diff
@@ -1,59 +1,342 @@
+const pickWeapon = () => { // Shows loading animation
+  document.getElementById('loader2').style.display = 'block';
+  setTimeout(realPickWeapon, 1000); // Delays the real pickWeapon function and calls it after 1.5 sec.
+};
+document.getElementById('weaponCard').addEventListener('click', pickWeapon);
+
+const realPickWeapon = () => { // The REAL function getting the values for the weapons
+  document.getElementById('loader2').style.display = 'none';
+  document.getElementById('weaponCard').removeEventListener('click', pickWeapon); // So the loader doesn't pops up in next click, it looks a bit weird if that happens
+
+  mystery.weapon = randomSelector(weapons); // Randomly selects a weapon
+      
+
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`;
+  document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight} pounds`;
+  weaponImage.src = mystery.weapon.image;
+
+  document.getElementById('weaponCard').onclick = realPickWeapon;
+};
+
+// PICK-ROOM-FUNCTION! This function will be invoked when you click on the room card.- in the similar way as pickKiller.
+const pickRoom = () => { // Shows loading animation
+  document.getElementById('loader3').style.display = 'block';
+  
+  setTimeout(realPickRoom, 1000); // Delays the real pickRoom function and calls it after 1.5 sec.
+};
+document.getElementById('roomCard').addEventListener('click', pickRoom);
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+const realPickRoom = () => { // The REAL function getting the values for the rooms
+  document.getElementById('loader3').style.display = 'none';
+  document.getElementById('roomCard').removeEventListener('click', pickRoom); // So the loader doesn't pops up in next click
 
-// This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.room = randomSelector(rooms); // Randomly selects a room
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+  document.getElementById('roomName').innerHTML = `${mystery.room.name}`;
+  roomImage.src = mystery.room.image;
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+  document.getElementById('roomCard').onclick = realPickRoom;
+};
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
```

Comment:
Your comments in this project makes it real easy to understand the code, keep it up!